src/main/java/com/ontotext/kafka/error/LogErrorHandler.java

```diff
@@ -70,4 +74,42 @@ private Properties getProperties(Map<String, ?> properties) {
 	public boolean isTolerable() {
 		return tolerance == ToleranceType.ALL;
 	}
+
+	private void resolvePropertiesFromEnvironment(Properties props) {
+		var envVars = System.getenv();
+		for (Map.Entry<String, String> entry : envVars.entrySet()) {
+			var key = entry.getKey();
+			if (key.startsWith(CONNECT_ENV_PREFIX)) {
+				key = key.replaceFirst("^CONNECT_CONSUMER_", "")
+					.replaceFirst("^CONNECT_PRODUCER_", "")
+					.replaceFirst("^" + CONNECT_ENV_PREFIX, "").replace("_", ".").toLowerCase();
+				var entryValue = entry.getValue();
+				entryValue = entryValue.replace("\\" + System.getProperty("line.separator", "\n"), "");
+				props.put(key, entryValue);
+			}
+		}
+	}
+
+	private boolean resolveProducerProperties(Map<String, ?> properties, Properties props) {
+		boolean producerPropertiesResolvedFromPassed = false;
+		for (Map.Entry<String, ?> entry : properties.entrySet()) {
+			var key = entry.getKey();
+			if (key.startsWith(PRODUCER_OVERRIDE_PROPERTY)) {
+				producerPropertiesResolvedFromPassed = true;
+				props.put(key.substring(PRODUCER_OVERRIDE_PROPERTY.length()), entry.getValue());
+			}
+		}
+		if (producerPropertiesResolvedFromPassed) {
```

Comment:
I don't get this change, shouldn't we add this property in any case?

Reply:
No it will resolve from env variables.